pya/aserver.py

```diff
@@ -225,93 +376,54 @@ def play(self, asig, onset: Union[int, float] = 0, out: int = 0, **kwargs):
+            asig_data = asig.sig.astype(self.backend.dtype)
+        else:
+            asig_data = asig.sig
+
+        event = AudioEvent(asig_data, rt_onset, out)
+        self.event_scheduler.schedule(event)
+
+        if "block" in kwargs and kwargs["block"]:
             if onset > 0:  # here really omset and not rt_onset!
                 _LOGGER.warning("blocking inactive with play(onset>0)")
             else:
                 time.sleep(asig.get_duration())
+
         return self
 
     def _play_callback(self, in_data, frame_count, time_info, flag):
         """callback function, called from pastream thread when data needed."""
-        tnow = self.block_time
-        self.block_time += self.block_duration
-        # self.block_cnt += 1  # TODO this will get very large eventually
-        # just curious - not needed but for time stability check
-        self.timejitter = time.time() - self.block_time
-        if self.timejitter > 3 * self.block_duration:
-            msg = f"Aserver late by {self.timejitter} seconds: block_time reset!"
-            _LOGGER.debug(msg)
-            self.block_time = time.time()
-        # to shortcut computing
-        if not self.srv_asigs or self.srv_onsets[0] > tnow:
-            return self.backend.process_buffer(self.empty_buffer)
-        elif self._stop:
-            self.srv_asigs.clear()
-            self.srv_onsets.clear()
-            self.srv_curpos.clear()
-            self.srv_outs.clear()
-            return self.backend.process_buffer(self.empty_buffer)
-        data = np.zeros((self.bs, self.channels), dtype=self.backend.dtype)
-        # iterate through all registered asigs, adding samples to play
-        dellist = []  # memorize completed items for deletion
-        t_next_block = tnow + self.bs / self.sr
-        for i, t in enumerate(self.srv_onsets):
-            if t > t_next_block:  # doesn't begin before next block
-                break  # since list is always onset-sorted
-            a = self.srv_asigs[i]
-            c = self.srv_curpos[i]
-            if t > tnow:  # first block: apply precise zero padding
-                io0 = int((t - tnow) * self.sr)
-            else:
-                io0 = 0
-            tmpsig = a.sig[c:c + self.bs - io0]
-            n, nch = tmpsig.shape
-            out = self.srv_outs[i]
-            # .reshape(n, nch) not needed as moved to play
-            data[io0:io0 + n, out:out + nch] += tmpsig
-            self.srv_curpos[i] += n
-            if self.srv_curpos[i] >= a.samples:
-                dellist.append(i)  # store for deletion
-        # clean up lists
-        for i in dellist[::-1]:  # traverse backwards!
-            del self.srv_asigs[i]
-            del self.srv_onsets[i]
-            del self.srv_curpos[i]
-            del self.srv_outs[i]
-        return self.backend.process_buffer(data * (self.backend.range * self.gain))
+        current_time = self.block_time
+        self.block_time += self.bs / self.sr
```

Comment:
I used `self.block_time += self.block_duration` to avoid the unnecessary recomputation of 
self.bs /self.sr (which usually remain unchanged) for performance reasons. what is the reason to undo this?